src/main/java/net/minestom/server/listener/CreativeInventoryActionListener.java

```diff
@@ -27,10 +29,16 @@ public static void listener(ClientCreativeInventoryActionPacket packet, Player p
         // Set item
         slot = (short) PlayerInventoryUtils.convertWindow0SlotToMinestomSlot(slot);
         PlayerInventory inventory = player.getInventory();
-        if (Objects.equals(inventory.getItemStack(slot), item)) {
+
+        CreativeInventoryActionEvent event = new CreativeInventoryActionEvent(player, slot, sentItem);
+        EventDispatcher.call(event);
+        final ItemStack setItem = event.getClickedItem();
+
+        if (Objects.equals(inventory.getItemStack(slot), sentItem) && Objects.equals(setItem, sentItem)) {
             // Item is already present, ignore
             return;
         }
-        inventory.setItemStack(slot, item);
+
+        inventory.setItemStack(slot, setItem);
```

Comment:
I believe this is not sufficient handling - If the event sets the item back to whats currently in the inventory (eg to prevent it), then minestom would not send the slot update in this case because the item in the inventory didnt change. But we must send a slot update here always.

Also - Perhaps the event should be cancellable and revert the item to server side version if cancelled?